test/e2e/specs/onboarding/signup__with-theme-premium.ts

```diff
@@ -0,0 +1,218 @@
+		} );
+
+		it( `Select WordPress.com ${ planName } plan`, async function () {
+			signupPickPlanPage = new SignupPickPlanPage( page );
+			newSiteDetails = await signupPickPlanPage.selectPlan( planName );
+		} );
+
+		it( 'See secure payment', async function () {
+			cartCheckoutPage = new CartCheckoutPage( page );
+			await cartCheckoutPage.validateCartItem( `WordPress.com ${ planName }` );
+		} );
+
+		it( 'Prices are shown in GBP', async function () {
+			const cartAmount = ( await cartCheckoutPage.getCheckoutTotalAmount( {
+				rawString: true,
+			} ) ) as string;
+			expect( cartAmount.startsWith( '£' ) ).toBe( true );
+		} );
+
+		it( 'Apply coupon', async function () {
+			originalAmount = ( await cartCheckoutPage.getCheckoutTotalAmount() ) as number;
+			await cartCheckoutPage.enterCouponCode( SecretsManager.secrets.testCouponCode );
+		} );
+
+		it( 'Apply coupon and validate purchase amount', async function () {
+			const newAmount = ( await cartCheckoutPage.getCheckoutTotalAmount() ) as number;
+
+			expect( newAmount ).toBeLessThan( originalAmount );
+			const expectedAmount = originalAmount * 0.99;
+
+			// Some currencies do not typically have decimal places.
+			// eg. USD would commonly have 2 decimal places, e.g. 12.34.
+			// In JPY or TWD there will be no decimal digits.
+			// Drop decimals so that the result won't be affected by the currency variation.
+			expect( newAmount ).toStrictEqual( expectedAmount );
+		} );
+
+		it( 'Enter billing and payment details', async function () {
+			const paymentDetails = DataHelper.getTestPaymentDetails();
+			await cartCheckoutPage.enterBillingDetails( paymentDetails );
+			await cartCheckoutPage.enterPaymentDetails( paymentDetails );
+		} );
+
+		it( 'Make purchase', async function () {
+			await cartCheckoutPage.purchase( { timeout: 90 * 1000 } );
+		} );
+
+		it( 'Skip upsell if present', async function () {
+			const selector = 'button[data-e2e-button="decline"]';
+			await page.waitForSelector( selector );
+
+			const locator = page.locator( selector );
+
+			try {
+				await locator.click( { timeout: 2 * 1000 } );
+			} catch {
+				// noop
+			}
+		} );
+
+		it( 'Click on the continue on Design Picker', async function () {
+			const startSiteFlow = new StartSiteFlow( page );
+
+			await startSiteFlow.clickButton( 'Continue' );
+		} );
+
+		it( 'Land in Home dashboard', async function () {
+			await page.waitForURL(
+				DataHelper.getCalypsoURL( `/home/${ newSiteDetails.blog_details.site_slug }` ),
+				{ timeout: 30 * 1000 }
+			);
+		} );
+
+		it( 'Checks the active theme', async function () {
+			const restAPIClient = new RestAPIClient(
+				{
+					username: testUser.username,
+					password: testUser.password,
+				},
+				newUserDetails.body.bearer_token
+			);
+
+			const theme = await restAPIClient.getActiveTheme( newSiteDetails.blog_details.blogid );
+
+			expect( theme ).toBe( `premium/${ themeSlug }` );
+		} );
```

Comment:
**question:kinda-blocking** - we aren't checking on the UI whether the theme that was selected is actually applied. Is that "enough" testing?

Reply:
I think so, we are interested in testing only the onboarding flow. We want to see that the API returns the correct theme for the site.

The UI for the active theme will change anyway soon.